src/amo/components/AddonMeta/index.js

```diff
@@ -51,8 +51,15 @@ export class AddonMetaBase extends React.Component<Props> {
     return (
       <div className="AddonMeta">
         <h3 className="visually-hidden">{i18n.gettext('Used by')}</h3>
-        <MetadataCard
-          metadata={[
+        <Link
```

Comment:
As mentioned above: there shouldn’t be a link here unless the addon is loaded. In your current code path: while an addon is loading this will link to ‘/‘, which is the homepage. That shouldn’t happen.